### PR TITLE
IDEMPIERE-5803 - ESC Shortcut Closes Tab and Pop-up Window at Once

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5803

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
  - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
  - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Comment by PeterTakacs300** [comment]
Summary:
- delay the execution of ON_CTRL_KEY after ON_CANCEL by adding echo: ON_CLOSE_WINDOW_SHORTCUT_EVENT (similarly as suggested by Heng Sin in the [Mattermost thread](https://mattermost.idempiere.org/idempiere/pl/yirwn1ns3fdsujanactsjhg4aa))
- improve IDesktop by isCloseTabWithShortcut methods
- the ON_CANCEL event sets isCloseTabWithShortcut to false
- tab is closed after ON_CTRL_KEY if isCloseTabWithShortcut is true

**Review by hengsin — CHANGES_REQUESTED** [review]
Fail with following test case:
- With default Alt-X only
- Open Product Info 
- Open lookup for vendor, press Esc to close lookup
- Press Alt-x to close product info tab and it is not working

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java [hunk 1]
```diff
@@ -2450,7 +2450,8 @@ else if(IDesktop.ON_CLOSE_WINDOW_SHORTCUT_EVENT.equals(event.getName())) {
 	 */
 	protected void onCancel() {
 		// do not allow to close tab for Events.ON_CTRL_KEY event
-		SessionManager.getAppDesktop().setCloseTabWithShortcut(false);
+		if(isUseEscForTabClosing)
+			SessionManager.getAppDesktop().setCloseTabWithShortcut(false);
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java:2454]: shouldn't the same check applicable to other onCancel changes as well ?

Comment (PeterTakacs300) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java:2454]: You're right, it should. I was able to reproduce the issue with opening a process modal dialog over an info window too.
I've added the logic to all types in the latest commit.

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**